reflex/config.py

```diff
@@ -257,21 +233,50 @@ class Config:  # pyright: ignore [reportIncompatibleVariableOverride]
     extra_overlay_function: str | None = None
 
     # List of plugins to use in the app.
-    plugins: list[Plugin] = []
+    plugins: list[Plugin] = dataclasses.field(default_factory=list)
 
     _prefixes: ClassVar[list[str]] = ["REFLEX_"]
 
-    def __init__(self, *args, **kwargs):
-        """Initialize the config values.
+
+@dataclasses.dataclass(kw_only=True, init=False)
+class Config(BaseConfig):
+    """The config defines runtime settings for the app.
+
+    By default, the config is defined in an `rxconfig.py` file in the root of the app.
+
+    ```python
+    # rxconfig.py
+    import reflex as rx
+
+    config = rx.Config(
+        app_name="myapp",
+        api_url="http://localhost:8000",
+    )
+    ```
+
+    Every config value can be overridden by an environment variable with the same name in uppercase.
+    For example, `db_url` can be overridden by setting the `DB_URL` environment variable.
+
```

Comment:
probably should update this while we're here
```suggestion
    Every config value can be overridden by an environment variable with the same name in uppercase and a REFLEX_ prefix.
    For example, `db_url` can be overridden by setting the `REFLEX_DB_URL` environment variable.

```